### DISCUSSION TITLE
Custom properties/dimensions not logged in AppInsights when logging custom events

### DISCUSSION THREAD

**Discussion by fgheysels** [arcus-azure/arcus.observability#191] [Q&A]
I log custom events in an Azure Function using the Arcus Observability methods like this:

```csharp
var customDimensions = new Dictionary<string, object>()
{
      ["VesselId"] = v.VesselId,
      ["Timestamp"] = v.Timestamp
};

log.LogEvent("VesselHeartbeatReceived", customDimensions);
```

I'd expect that all my  'VesselHeartbeatReceived' custom events now contain 2 custom properties, but that doesn't seem to be the case.

Given this KQL query:

```kql
customEvents 
| where name=="VesselHeartbeatReceived"
| project name,  customDimensions["VesselId"], timestamp
| order by timestamp desc 
```

I get this output:

![image](https://user-images.githubusercontent.com/3605786/110515928-37747580-8109-11eb-8b33-ce8f49f48a90.png)

As you can see, some events have custom dimensions, others do not.
When I go somewhat further back in time, I can again see some events with those custom properties:

![image](https://user-images.githubusercontent.com/3605786/110516052-5b37bb80-8109-11eb-9bdc-bae3f001e2f0.png)

How to explain this behavior ?  A bug in Arcus.Observability ?  Sampling by AppInsights ?

**Comment by tomkerkhove** [comment]
Hm that's odd, was it with a given change or not?

**Reply by fgheysels** [reply]
No, I just added the logging of custom events to that function.

Atm, all events that are registered, have those custom metrics though at first sight ....

**Reply by tomkerkhove** [reply]
Hm odd 🤔 So now and then they are added and sometimes they are not?

**Reply by fgheysels** [reply]
This is no longer occuring :?
I mean, when I perform a query to retrieve customevents now, all events have my custom dimensions.

**Reply by tomkerkhove** [reply]
The cloud, it is magical

![](https://media.tenor.com/images/0086c213951436b5bece61e28900c72a/tenor.gif)

**Reply by tomkerkhove** [reply]
Let's maybe mark it as resolved and bump if it happens again?

**Reply by fgheysels** [reply]
Weird, it is happening again.  I've changed my custom properties, redeployed the function: no custom dimensions available.
